docs/sql-migration-guide-upgrade.md

```diff
@@ -9,6 +9,8 @@ displayTitle: Spark SQL Upgrading Guide
 
 ## Upgrading From Spark SQL 2.4 to 3.0
 
+  - In Spark version 2.4 and earlier, accepted format of decimals parsed from JSON is an optional sign ('+' or '-'), followed by a sequence of zero or more decimal digits, optionally followed by a fraction, optionally followed by an exponent. Any commas were removed from the input before parsing. Since Spark 3.0, format varies and depends on locale which can be set via JSON option `locale`. The default locale is `en-US`. To switch back to previous behavior, set `spark.sql.legacy.decimalParsing.enabled` to `true`.
```

Comment:
Is there any performance regression with `DecimalFormat`?

Reply:
I have the same question. Do we need the `DecimalFormat` when locale is `en-US`?

Reply:
> Is there any performance regression with DecimalFormat?

I haven't benchmarked the changes. Looking at `JSONBenchmarks`, we even don't check decimals there.

> Do we need the DecimalFormat when locale is en-US?

No, we don't need it.

Reply:
since the default value is `en-US`, can we skip `DecimalFormat` when locale is `en-US`? Then there is nothing changes by default, and we don't even need a config.

Reply:
I removed SQL config and record in the migration guid. Also I applied `DecimalFormat` only for not `en-US` locales.